### PR TITLE
Update: BardLite - Allow any key in dimension value

### PR DESCRIPTION
**Pull request by balajimanoharan**
Update: BardLite - Allow any key in dimension value beyond just id and description

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/data/addon/mirage/routes/bard-lite.js [hunk 1]
```diff
@@ -187,13 +187,18 @@ export default function(
...
+          dimensionValues.map(value => {
             // TODO figure out why Object.assign refuses to work in Phantom even with Babel polyfill
-            assign({}, currentRow, {
-              [`${dimension}|id`]: value.id,
+            let newValue = assign({}, currentRow, {
               [`${dimension}|desc`]: value.description
-            })
-          )
+            });
+            const keys = Object.keys(value).filter(key => key !== 'description');
```
Comment (kevinhinterlong) [inline packages/data/addon/mirage/routes/bard-lite.js]: Maybe just `.map` with an `if(key === description)`

File: packages/data/addon/mirage/routes/bard-lite.js [hunk 2]
```diff
@@ -188,10 +187,12 @@ export default function(
 
         return newRows.concat(
           dimensionValues.map(value =>
-            // TODO figure out why Object.assign refuses to work in Phantom even with Babel polyfill
-            assign({}, currentRow, {
-              [`${dimension}|id`]: value.id,
-              [`${dimension}|desc`]: value.description
+            Object.keys(value).map(key => {
```
Comment (mattgraham1995) [inline packages/data/addon/mirage/routes/bard-lite.js]: We're using map but not returning anything in the map callback. Is that the functionality we want?

Comment (kevinhinterlong) [reply packages/data/addon/mirage/routes/bard-lite.js]: In addition to this, it should map each  key to an object then reduce to one object and use assign. Or just replace the map with forEach

Comment (balajimanoharan) [reply packages/data/addon/mirage/routes/bard-lite.js]: ummm.. ya.. my bad.. fixed now.

File: packages/data/addon/mirage/routes/bard-lite.js [hunk 3]
```diff
@@ -187,13 +186,16 @@ export default function(
...
-          )
+          dimensionValues.map(value => {
+            Object.keys(value).forEach(key => {
+              if (key === 'description') {
+                currentRow[`${dimension}|desc`] = value[key];
+              } else {
+                currentRow[`${dimension}|${key}`] = value[key];
+              }
+            });
+            return currentRow;
```
Comment (mattgraham1995) [inline packages/data/addon/mirage/routes/bard-lite.js]: Won't this result in multiple copies of `currentRow` being returned in the `dimensionValues.map`?

**Pull request merged**